Library/Homebrew/api/formula.rb

```diff
@@ -25,6 +25,8 @@ def self.formula_json(name)
 
       sig { params(name: String, download_queue: T.nilable(DownloadQueue)).void }
       def self.fetch_formula_json!(name, download_queue: nil)
+        raise "#{name}: attempted to fetch formula JSON!" if ENV["HOMEBREW_NO_FETCH_API_FORMULA_JSON"]
```

Comment:
At the moment, this will probably break many commands like `brew deps`, because the arg parser automatically fetches the JSON files for any named arguments (except in a few special cases like `brew outdated`)

Is the intention to change that behavior, or should this be moved to e.g. `Formulary::FromAPILoader::load_from_api`?